[web-animations] `stroke-color` doesn't work with `transition: all`

nt1m opened the pull request:
#### 3525ae62d545bb3727b8778e942718d17cb67333
<pre>
[web-animations] `stroke-color` doesn&apos;t work with `transition: all`
<a href="https://bugs.webkit.org/show_bug.cgi?id=277370">https://bugs.webkit.org/show_bug.cgi?id=277370</a>
<a href="https://rdar.apple.com/132837478">rdar://132837478</a>

Reviewed by Matthieu Dubet.

281570@main forgot to add the property to `RenderStyle::conservativelyCollectChangedAnimatableProperties`.

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/fill-stroke/animation/stroke-color-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/fill-stroke/animation/stroke-color-interpolation.html: Added.
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):

Canonical link: <a href="https://commits.webkit.org/281619@main">https://commits.webkit.org/281619@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2edba3d4488ba905f69fe667ca5e7a76c624dc3f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60466 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39817 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13036 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64389 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11003 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62596 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47492 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11230 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/48924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7647 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62497 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37095 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/52384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/29766 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33792 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/9615 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9918 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/55665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9910 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66120 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4399 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9724 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4423 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52361 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/56458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [💥 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13441 "An unexpected error occured. Recent messages:OS: Ventura (13.6.7), Xcode: 14.3; Checked out pull request; Reviewed by Matthieu Dubet; Compiled WebKit (warnings); Exiting early after 60 failures. 14273 tests run. 60 failures") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3659 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9086 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/35629 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/36711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/37802 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on LayoutTests/imported/w3c/web-platform-tests/css/fill-stroke/animation/stroke-color-interpolation-expected.txt:
```diff
@@ -0,0 +1,174 @@
+FAIL Web Animations: property <stroke-color> from [unset] to [green] at (0.3) should be [rgba(0, 128, 0, 0.3)] assert_equals: expected "rgba ( 0 , 128 , 0 , 0.3 ) " but got "rgb ( 167 , 205 , 167 ) "
+FAIL Web Animations: property <stroke-color> from [unset] to [green] at (0.6) should be [rgba(0, 128, 0, 0.6)] assert_equals: expected "rgba ( 0 , 128 , 0 , 0.6 ) " but got "rgb ( 95 , 172 , 95 ) "
+PASS Web Animations: property <stroke-color> from [unset] to [green] at (1) should be [rgb(0, 128, 0)]
+FAIL Web Animations: property <stroke-color> from [unset] to [green] at (1.5) should be [rgb(0, 192, 0)] assert_equals: expected "rgb ( 0 , 192 , 0 ) " but got "rgb ( 0 , 73 , 0 ) "
```
We don't like `unset`

nt1m replied:
oops I need to fix that

nt1m committed:
[web-animations] `stroke-color` doesn't work with `transition: all`
https://bugs.webkit.org/show_bug.cgi?id=277370
rdar://132837478

Reviewed by Matthieu Dubet.

281570@main forgot to add the property to `RenderStyle::conservativelyCollectChangedAnimatableProperties`.

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/fill-stroke/animation/stroke-color-interpolation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/fill-stroke/animation/stroke-color-interpolation.html: Added.
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):

Canonical link: https://commits.webkit.org/281619@main
